app/scripts/lib/transaction/institutional-snap/deferred-publication-hooks.ts

```diff
@@ -0,0 +1,186 @@
+  never
+>;
+
+type DeferrableTransactionAccount = {
+  options: {
+    custodian: {
+      deferPublication: boolean;
+    };
+  };
+};
+
+type DeferredPublicationControllerState = Record<string, never>;
+
+const metadata = {};
+
+export class DeferredPublicationController extends BaseController<
+  'DeferredPublicationController',
+  DeferredPublicationControllerState,
+  DeferredPublicationHookMessenger
+> {
+  constructor({ messenger }: { messenger: DeferredPublicationHookMessenger }) {
+    super({
+      messenger,
+      name: 'DeferredPublicationController',
+      state: {},
+      metadata,
+    });
+  }
+
+  private async handleSnapRequest(args: SnapRPCRequest) {
+    const response = await this.messagingSystem.call(
+      'SnapController:handleRequest',
+      args,
+    );
+    return response as InstitutionalSnapResponse;
+  }
+
+  private async shouldDeferPublication(transactionMeta: TransactionMeta) {
+    const account = (await this.messagingSystem.call(
+      'AccountsController:getAccountByAddress',
+      transactionMeta.txParams.from as string,
+    )) as unknown as DeferrableTransactionAccount;
+
+    return account?.options.custodian?.deferPublication;
+  }
+
+  async deferPublicationHook(
+    transactionController: TransactionController,
+    transactionMeta: TransactionMeta,
+  ): Promise<boolean> {
+    const shouldDefer = await this.shouldDeferPublication(transactionMeta);
+
+    if (shouldDefer) {
+      const searchParams: InstitutionalSnapRequestSearchParameters = {
+        from: transactionMeta.txParams.from as string,
+        to: transactionMeta.txParams.to as string,
+        value: transactionMeta.txParams.value as string,
+        data: transactionMeta.txParams.data as string,
+        chainId: transactionMeta.chainId as string,
+      };
+
+      const snapGetMutableTransactionParamsPayload: SnapRPCRequest = {
+        snapId,
+        origin: 'metamask',
+        handler: HandlerType.OnRpcRequest,
+        request: {
+          method: 'transactions.getMutableTransactionParameters',
+          params: searchParams,
+        },
+      };
+
+      const snapResponse = await this.handleSnapRequest(
+        snapGetMutableTransactionParamsPayload,
+      );
+
+      const hash = snapResponse.transaction.transactionHash;
+
+      transactionController.updateCustodialTransaction(transactionMeta.id, {
+        status: TransactionStatus.submitted,
+        hash,
+        nonce: snapResponse.transaction.nonce,
+        gasLimit: snapResponse.transaction.gasLimit,
+        maxFeePerGas: snapResponse.transaction.maxFeePerGas,
+        maxPriorityFeePerGas: snapResponse.transaction.maxPriorityFeePerGas,
+        type: snapResponse.transaction.type as TransactionEnvelopeType,
+      });
+      return false;
+    }
+    return true;
+  }
+
+  async beforeCheckPendingTransactionHook(transactionMeta: TransactionMeta) {
+    const shouldDefer = await this.shouldDeferPublication(transactionMeta);
```

Comment:
Minor, could we just immediately return and negate without the variable as the method name is explicit?

Reply:
Addressed [here](https://github.com/MetaMask/metamask-extension/pull/30180/files#diff-fc1652c483153bd55fe46a633567c33518fd9310f5ad683244926d8e8e100f04R177)